pkg/broker/amqp/amqp.go

```diff
@@ -14,13 +15,22 @@ type Client struct {
 	QueueName        string
 }
 
+//getEnvWithError func returns error if the env is not found
+func getEnvWithError(env string) (string, error) {
+	envValue, found := os.LookupEnv(env)
+	if !found {
+		return "", fmt.Errorf("%s env not found", env)
+	}
+	return envValue, nil
+}
+
 //NewClient creates a new Client for amqp broker
 func NewClient() (broker.Broker, error) {
-	connectionString, err := shared.GetEnvWithError("BROKER_CONNECTION_STRING")
+	connectionString, err := getEnvWithError("BROKER_CONNECTION_STRING")
```

Comment:
I think that both `connectionString` and `queue` can be constructor parameters.
I will be easier to test, to reuse in another projet/tool and it will remove the code duplication for `getEnvWithError` :)